web3/utils/validation.py

```diff
@@ -23,8 +23,8 @@ def validate_address(address):
     Helper function for validating an address
     """
     if not is_address(address):
+        validate_address_checksum(address)
```

Comment:
`validate_address_checksum` to raises a ValueError while `validate_address` used to raise a TypeError. Because `is_address` now can return false due to wrong checksum, it was an attempt to catch whether that was because the checksum failed or otherwise to throw the ValueError vs TypeError.

When calling `validate_address`, if the address has a bad checksum, it will never get to `validate_address_checksum` now. It will simply throw the generic ValueError without any details about checksum.

I know the `validate_address_checksum` under the block of code looks strange, but I think it gives us the desired functionality.